toolbox/relnotes/main.go

```diff
@@ -0,0 +1,606 @@
+	if err != nil {
+		log.Printf("failed to get release commits for %s: %s", branchRange, err)
+		return
+	}
+
+	// Parse release related PR ids from the release commits
+	commitPRs, err := parsePRFromCommit(releaseCommits)
+	if err != nil {
+		log.Printf("failed to parse release commits: %s", err)
+		return
+	}
+
+	// Get number-issue mapping for issues in the repository
+	issues, err := u.ListAllIssues(client, *owner, *repo)
+	if err != nil {
+		log.Printf("failed to list all issues from %s: %s", *repo, err)
+		return
+	}
+	issueMap := make(map[int]*github.Issue)
+	for _, i := range issues {
+		issueMap[*i.Number] = i
+	}
+
+	// Get release note PRs by examining release-note label on commit PRs
+	releasePRs := make([]int, 0)
+	for _, pr := range commitPRs {
+		if u.HasLabel(issueMap[pr], "release-note") {
+			releasePRs = append(releasePRs, pr)
+		}
+	}
+
+	// Generating release note...
+	log.Printf("Generating release notes...")
+
+	prFile, err := os.Create(prFileName)
+	if err != nil {
+		log.Printf("failed to create release note file %s: %s", prFileName, err)
+		return
+	}
+
+	// Bootstrap notes for minor (new branch) releases
+	if *full || u.IsVer(releaseTag, "dotzero") {
+		draftURL := u.GithubRawURL + *owner + "/features/master/" + *branch + "/release-notes-draft.md"
+		changelogURL := u.GithubRawURL + *owner + "/" + *repo + "/master/CHANGELOG.md"
+		minorRelease(prFile, releaseTag, draftURL, changelogURL)
+	} else {
+		patchRelease(prFile, startTag, releasePRs, issueMap)
+	}
+
+	prFile.Close()
+
+	// Start generating markdown file
+	log.Printf("Preparing layout...")
+
+	mdFile, err := os.Create(*mdFileName)
+	if err != nil {
+		log.Printf("failed to create release note markdown file %s: %s", *mdFileName, err)
+		return
+	}
+
+	// Create markdown file body with hardcoded kubernetes URLs
+	docURL := "https://docs.k8s.io"
+	exampleURL := "https://releases.k8s.io/" + *branch + "/examples"
+	createBody(client, mdFile, releaseTag, *branch, docURL, exampleURL, *releaseTars)
+
+	// Copy (append) the pull request notes into the output markdown file
+	prFile, _ = os.Open(prFileName)
+	_, err = io.Copy(mdFile, prFile)
+	if err != nil {
+		log.Printf("failed to copy from PR file to release note markdown file: %s", err)
+	}
+	err = mdFile.Sync()
+	if err != nil {
+		log.Printf("failed to copy from PR file to release note markdown file: %s", err)
+	}
+
+	prFile.Close()
+
+	if *preview {
+		// If in preview mode, get the pending PRs
+		err = getPendingPRs(client, mdFile, *owner, *repo, *branch)
+		if err != nil {
+			log.Printf("failed to get pending PRs: %s", err)
+			return
+		}
+	}
+	mdFile.Close()
+
+	if *htmlizeMD {
+		// Make users and PRs linkable
+		// Also, expand anchors (needed for email announce())
+		k8sGithubURL := "https://github.com/kubernetes/kubernetes"
+		_, err = u.Shell("sed", "-i", "-e", "s,#\\([0-9]\\{5\\,\\}\\),[#\\1]("+k8sGithubURL+"/pull/\\1),g",
```

Comment:
Try not rely on bash command in golang program. You can do something like [here](https://github.com/istio/istio/blob/master/tests/e2e/framework/kubernetes.go#L327)